packages/components/src/font-size-picker/index.js

```diff
@@ -6,51 +6,108 @@ import { map } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { withInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
+import Dashicon from '../dashicon';
+import BaseControl from '../base-control';
 import Button from '../button';
-import ButtonGroup from '../button-group';
+import Dropdown from '../dropdown';
 import RangeControl from '../range-control';
+import { NavigableMenu } from '../navigable-container';
+
+function FontSizePicker( {
+	fallbackFontSize,
+	fontSizes = [],
+	instanceId,
+	onChange,
+	value,
+	withSlider,
+} ) {
+	const onChangeValue = ( event ) => {
+		const newValue = event.target.value;
+		if ( newValue === '' ) {
+			onChange( undefined );
+			return;
+		}
+		onChange( Number( newValue ) );
+	};
+
+	const currentFont = fontSizes.find( ( font ) => font.size === value );
+	const labelId = `components-font-size-picker-label-${ instanceId }`;
 
-export default function FontSizePicker( { fontSizes = [], fallbackFontSize, value, onChange } ) {
 	return (
-		<Fragment>
+		<BaseControl
+			label={ __( 'Font Size' ) }
```

Comment:
The BaseControl renders a `<label>` element with a `for` id that is not associated to any element. It's an orphaned label which basically doesn't do anything. So, the visible text "Font Size" shouldn't be a `<label>` element.

Reply:
> The BaseControl renders a `<label>` element with a `for` id that is not associated to any element. It's an orphaned label which basically doesn't do anything. So, the visible text "Font Size" shouldn't be a `<label>` element.

@afercia Noting this after my comment at https://github.com/WordPress/gutenberg/pull/9802#discussion_r257249492 . If I understand correctly, the issue was more that a `label` element was being rendered, but without the necessary `for` attribute? Am I correct in thinking that there should either be a `for` attribute assigned, or no label (i.e. text) at all? As merged, the implementation was changed to render a `span` in place of the label, which doesn't feel quite right to me.

![image](https://user-images.githubusercontent.com/1779930/52862279-e7ca4800-3102-11e9-813c-5b41ed47654c.png)

Reply:
Yep I've just commented on https://github.com/WordPress/gutenberg/pull/10925#discussion_r257252993 using the same screenshot :D 

When the font size picker was changed to a custom component, the label was an orphaned label. 

> Am I correct in thinking that there should either be a `for` attribute assigned

The size picker custom component is made of buttons so a label with a `for` attribute wouldn't be appropriate.

> or no label (i.e. text) at all?

I guess the text is necessary for visual purposes.

For the records: I wasn't happy with the new size picker, it's non-native and its semantics and interaction are problematic. That change was introduced without even asking for some initial accessibility feedback. Alas, this is what happens when visual design considerations take priority on semantics and accessibility.

Reply:
Could the `label` element have directed to the `button` rendered by the custom dropdown (what appears in the screenshot above as "Normal" dropdown-looking element)?

Separately, considering how we might help avoid this scenario in the future: It would be fairly trivial to create an ESLint rule which forbids a `BaseControl` that includes `label` prop but omits `id`. One problem with this may be, as you point out, that a visual text may be appropriate as long as the inputs assign `aria-label`? Scanning those child elements would be more difficult to enforce. Arguably in those scenarios I might suggest the visual label be rendered as just another child, not using the `label` prop. That way, we can support both cases and ensure that `label` prop, if present, must have a corresponding `id`.

By above, what I mean is that both of these would be valid:

```jsx
<BaseControl label="Font Size" id="my-input">
	<input id="my-input" />
</BaseControl>
```

```jsx
<BaseControl>
	<span>Font Size</span>
	<input aria-label="Font Size" />
</BaseControl>
```

And the following invalid (a case very similar to #10925):

```jsx
<BaseControl label="Font Size">
	<input id="my-input" />
</BaseControl>
```

Reply:
> I might suggest the visual label be rendered as just another child, not using the label prop.

Seems the most appropriate thing to do!